src/mds/MDLog.cc

```diff
@@ -847,7 +847,9 @@ void MDLog::_recovery_thread(MDSInternalContextBase *completion)
   } else if (read_result != 0) {
     mds->clog->error() << "failed to read JournalPointer: " << read_result
                        << " (" << cpp_strerror(read_result) << ")";
+    mds->mds_lock.Lock();
     mds->damaged();
+    mds->mds_lock.Unlock();
```

Comment:
If damaged() is called from multiple contexts it should assert that the current thread holds the mds_lock.